src/scripts/modules/search/advanced/advanced.coffee

```diff
@@ -69,11 +68,12 @@ define (require) ->
       query = @formatQuery(values)
 
       @search(query)
-      if !query
-        $("#alert-danger").show()
 
-    search: (query) ->
-      router.navigate("search?q=#{query}", {trigger: true})
+    search: (query,values) ->
```

Comment:
Since validation is a separate action than a search, why don't we make a separate `validateSearchable` function for the query values. At the same time, we can take advantage of using the full data structure of `values` instead of `query` to validate. Hint: `values.filter(function(field) { return field.value !== "" }).every(function(field) { return field.name === "sort"})` might help you determine if a query is searchable or not